### ISSUE TITLE
Cannot build on latest xcode 15 b5 for ios 17

### ISSUE THREAD

**Issue by currantdev** [pichillilorenzo/flutter_inappwebview#1735]
- [x] I have read the [Getting Started](https://inappwebview.dev/docs/intro/) section
- [x] I have already searched for the same problem

## Environment

| Technology      | Version       |
|-----------------| ------------- |
| Flutter version |        3.7.12       |
| Plugin version  |         6.0.0 b24      |
| Android version |               |
| iOS version     |        17.0       |
| macOS version   |      sonoma public beta 1         |
| Xcode version   |        15 beta 5       |

Device information: iphone

## Description
Failed to build iOS app
Error (Xcode): type argument 'nw_proxy_config_t' (aka 'struct nw_proxy_config *') is neither an Objective-C object nor a block
type
/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Li
brary/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h:116:46

Parse Issue (Xcode): Could not build module 'WebKit'
/build/ios/Release-iphoneos/flutter_inappwebview/flutter_inappwebview.framework/Headers
/flutter_inappwebview-Swift.h:286:8

Encountered error while building for device.

**Expected behavior:** 
run or build ios app

**Current behavior:** 
erro while building 

## Steps to reproduce
Run on latest xcode version

**Comment by lqvv** [comment]
any suggestion ？

**Comment by Takabrycheri** [comment]
Same problem for me

**Comment by zzz2016** [comment]
Same problem

**Comment by Helios030** [comment]
Same problem

**Comment by fumin65** [comment]
Same problem

**Comment by Takabrycheri** [comment]
Same for Xcode 15 b6 (although quite obvious)

**Comment by abdulazeem4** [comment]
Same error here

**Comment by dizz** [comment]
same issue here too

**Comment by abdulazeem4** [comment]
Did someone find a work around/fix?

**Comment by dizz** [comment]
I commented out the dependency and related code for now. Tried to use the 6.x.x release but issue persisted.

**Comment by realitymolder** [comment]
Same issue here...

**Comment by jkaae93** [comment]
iOS 17 will be released as early as September 13, 2023.
I have less than a month to respond to iOS 17, is there a solution?

**Comment by erickyim** [comment]
**WARNING: NOT TESTED**
workaround 1): in framework xcodeproj `Build Phrases -> Compile Sources`，select all sources，delete `-DOS_OBJECT_USE_OBJC=0` in `Complier Flags`

workaround 2):  open ` /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h`, change `__IPHONE_OS_VERSION_MAX_ALLOWED` to `180000`

**Comment by dizz** [comment]
Received this from Erick Yim (not showing on the thread?!)

> WARNING: NOT TESTED
workaround 1): in framework xcodeproj Build Phrases -> Compile Sources，select all sources，delete -DOS_OBJECT_USE_OBJC=0 in Complier Flags

> workaround 2): open /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h, change __IPHONE_OS_VERSION_MAX_ALLOWED to 180000

workaround 2 has me happy for now! 😊

**Comment by erickyim** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by currantdev** [comment]
I can confirm -> workaround 2 is working

**Comment by istiak-sourav** [comment]
Same issue

**Comment by realitymolder** [comment]
> I can confirm -> workaround 2 is working

Same, thank you very everyone. Hope this issue will be fixed in the next beta.

**Comment by istiak-sourav** [comment]
Workaround 2 is working.
Thanks man, you're a life saver <3

**Comment by jay-7span** [comment]
Not working for me In my case the if condition is commented.
`#if ((TARGET_OS_OSX && __MAC_OS_X_VERSION_MAX_ALLOWED >= 140000) \
    || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 170000) \
    || (TARGET_OS_WATCH && __WATCH_OS_VERSION_MAX_ALLOWED >= 100000) \
    || (TARGET_OS_TV && __TV_OS_VERSION_MAX_ALLOWED >= 170000))
/*! @abstract Gets or sets the proxy configurations to be used to override networking in all WKWebViews that use this WKWebsiteDataStore.
 @discussion Changing the proxy configurations might interupt current networking operations in any WKWebView that use this WKWebsiteDataStore,
 so it is encouraged to finish setting the proxy configurations before starting any page loads.
*/
@property (nullable, nonatomic, copy) NSArray<nw_proxy_config_t> *proxyConfigurations NS_REFINED_FOR_SWIFT API_AVAILABLE(macos(14.0), ios(17.0));
#endif`

**Comment by demoYang** [comment]
mark

**Comment by ssly1997** [comment]
same issue

**Comment by currantdev** [comment]
Remember guys that every new xcode build (like latest beta 8) you have to made this change again.
I recommend workaround 2, because its working each time.
so Line 113 of file  **/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h**

Change from:
`    || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 170000) \
`
to:
`    || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 180000) \
`

**Comment by snipd-mikel** [comment]
I tried to do Workaround 2 but couldn't edit the file: It said that the file couldn't be written. I have tried with `sudo` as well but no luck. Has anyone experienced the same issue?

**Comment by ssly1997** [comment]
> I tried to do Workaround 2 but couldn't edit the file: It said that the file couldn't be written. I have tried with `sudo` as well but no luck. Has anyone experienced the same issue?

Using a text editor instead of Xcode

**Comment by snipd-mikel** [comment]
I have tried both vscode and vim and no luck. Which one did you use?

**Comment by ssly1997** [comment]
> I have tried both vscode and vim and no luck. Which one did you use?

Default TextEditor

**Comment by snipd-mikel** [comment]
> > I have tried both vscode and vim and no luck. Which one did you use?
> 
> Default TextEditor

Thanks that helped! I was getting crazy! What a mess updating to iOS17 😅

**Comment by techwithsam** [comment]
> 180000



> > > I have tried both vscode and vim and no luck. Which one did you use?
> > 
> > 
> > Default TextEditor
> 
> Thanks that helped! I was getting crazy! What a mess updating to iOS17 😅

How did you save the file after editing it on Textedit because what's i'm get is i don't have permission to save file even after explicitly giving permission to read and write.

**Comment by realitymolder** [comment]
> > 180000
> 
> > > > I have tried both vscode and vim and no luck. Which one did you use?
> > > 
> > > 
> > > Default TextEditor
> > 
> > 
> > Thanks that helped! I was getting crazy! What a mess updating to iOS17 😅
> 
> How did you save the file after editing it on Textedit because what's i'm get is i don't have permission to save file even after explicitly giving permission to read and write.

you should change the premission for this file but sometimes that is not enough. If that is the case use vim to edit this textfile and use sudo before the vim
so you will have:
`sudo vim fileName`
this way it will work 10000%

**Comment by techwithsam** [comment]
> > > 180000
> > 
> > 
> > > > > I have tried both vscode and vim and no luck. Which one did you use?
> > > > 
> > > > 
> > > > Default TextEditor
> > > 
> > > 
> > > Thanks that helped! I was getting crazy! What a mess updating to iOS17 😅
> > 
> > 
> > How did you save the file after editing it on Textedit because what's i'm get is i don't have permission to save file even after explicitly giving permission to read and write.
> 
> you should change the premission for this file but sometimes that is not enough. If that is the case use vim to edit this textfile and use sudo before the vim so you will have: `sudo vim fileName` this way it will work 10000%

So i need to download vim on my laptop? Because i don't have vim installed on my system

**Comment by realitymolder** [comment]
> > > > 180000
> > > 
> > > 
> > > > > > I have tried both vscode and vim and no luck. Which one did you use?
> > > > > 
> > > > > 
> > > > > Default TextEditor
> > > > 
> > > > 
> > > > Thanks that helped! I was getting crazy! What a mess updating to iOS17 😅
> > > 
> > > 
> > > How did you save the file after editing it on Textedit because what's i'm get is i don't have permission to save file even after explicitly giving permission to read and write.
> > 
> > you should change the premission for this file but sometimes that is not enough. If that is the case use vim to edit this textfile and use sudo before the vim so you will have: `sudo vim fileName` this way it will work 10000%
> 
> So i need to download vim on my laptop? Because i don't have vim installed on my system

If you are on mac you have vim built-in

**Comment by simonurzua** [comment]
Is this an error that can be fixed on the library itself? Should be fixed by Apple? Do you know what's the root cause?

**Comment by MazEbeid** [comment]
If you are struggling to edit, open terminal and run open open /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/

Then find WKWebsiteDataStore.h in finder and then right click open with TextEdit. Edit the file and save it again

**Comment by 9rganizedchaos** [comment]
If there's someone who are struggling with edit the `WKWebsiteDataStore.h` file,
first modify file access permission, refering this [document](https://support.apple.com/ko-kr/guide/mac-help/mchlp1203/mac)!

**Comment by TabooSun** [comment]
I found a way without hacking around with the `WKWebsiteDataStore.h`. Thanks to an article over here: https://juejin.cn/post/7268228858762854400

The solution that works for me is to edit the `podspec` of this plugin by adding this:
```ruby
  s.platforms = { :ios => '11.0' }
```

I have forked this project and made the necessary change in which you can just use by changing the `pubspec.yaml` to:
```
  flutter_inappwebview:
    git:
      url: https://github.com/Estrelio/flutter_inappwebview.git
      ref: fix-xcode-17
```

Take note that I made this change on top of the `master` branch which I believe it's version `6.0.0-beta.25`. If you are not using this version then you may need to change it your own.

And also, if you want to support iOS version below 11, it probably won't work as well.

**Comment by vixez** [comment]
@TabooSun 
Thanks, seems to work.
For anyone interested in the `5.7.2+3` version of this, I made a fork with this fix
```
  flutter_inappwebview:
    git:
      url: https://github.com/vixez/flutter_inappwebview.git
      ref: fix-xcode-17
```

**Comment by alexandr-efimov** [comment]
it works, but it is really...temporal hopefully :)

**Comment by TheMelody** [comment]
> 请记住，每个新的 xcode 版本（如最新的 beta 8）都必须再次进行此更改。我推荐解决方法 2，因为它每次都有效。所以文件 /**Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS113.17.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h** 的第 0 行
> 
> 从： 更改为： ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 170000) \` ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 180000) \`

cool

**Comment by durga-nannamsetty** [comment]
Thanks @vixez. It worked. But guys I am facing another issue after using the fork. I am new to flutter. So please give your inputs to fix this.

> youtube_player_flutter 8.1.2 depends on flutter_inappwebview ^5.7.2+3 and no versions of youtube_player_flutter match >8.1.2 <9.0.0,
  youtube_player_flutter ^8.1.2 requires flutter_inappwebview from hosted.

**Comment by crasowas** [comment]
> I found a way without hacking around with the `WKWebsiteDataStore.h`. Thanks to an article over here: https://juejin.cn/post/7268228858762854400
> 
> The solution that works for me is to edit the `podspec` of this plugin by adding this:
> 
> ```ruby
>   s.platforms = { :ios => '11.0' }
> ```
> 
> I have forked this project and made the necessary change in which you can just use by changing the `pubspec.yaml` to:
> 
> ```
>   flutter_inappwebview:
>     git:
>       url: https://github.com/Estrelio/flutter_inappwebview.git
>       ref: fix-xcode-17
> ```
> 
> Take note that I made this change on top of the `master` branch which I believe it's version `6.0.0-beta.25`. If you are not using this version then you may need to change it your own.
> 
> And also, if you want to support iOS version below 11, it probably won't work as well.

Thanks, it works.

I have tested many iOS versions and found that the minimum supported version is iOS 6.0.

What's confusing though is why `core.platform = :ios, '8.0'` didn't take effect?

**Comment by TabooSun** [comment]
> Thanks @vixez. It worked. But guys I am facing another issue after using the fork. I am new to flutter. So please give your inputs to fix this.
> 
> > youtube_player_flutter 8.1.2 depends on flutter_inappwebview ^5.7.2+3 and no versions of youtube_player_flutter match >8.1.2 <9.0.0,
> > youtube_player_flutter ^8.1.2 requires flutter_inappwebview from hosted.

This is because the package `youtube_player_flutter` you are using doesn't depend on the git dependency you are using.

You have 2 ways to address this:
1. Clone the `youtube_player_flutter` and update its `pubspec.yaml` to use the fork.
2. Use [dependency_overrides (not recommended)](https://dart.dev/tools/pub/dependencies#dependency-overrides)

**Comment by ZhiJieZhang1** [comment]
> > 请记住，每个新的 xcode 版本（如最新的 beta 8）都必须再次进行此更改。我推荐解决方法 2，因为它每次都有效。所以文件 /**Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS113.17.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h** 的第 0 行
> > 从： 更改为： ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 170000) \` ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 180000) \`
> 
> cool

WKWebsiteDataStore.h  文件不让修改，怎么办的？

**Comment by TheMelody** [comment]
> > > 请记住，每个新的 xcode 版本（如最新的 beta 8）都必须再次进行此更改。我推荐解决方法 2，因为它每次都有效。所以文件 /**Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS113.17.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h** 的第 0 行
> > > 从： 更改为： ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 170000) \` ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 180000) \`
> > 
> > 
> > cool
> 
> @TheMelody WKWebsiteDataStore.h 文件不让修改，怎么办的？

![01EF4D4D](https://github.com/pichillilorenzo/flutter_inappwebview/assets/9635441/bbe40a11-ec0d-4ed9-b58c-f6649e681926)  修改文件可读可写即可，你这......

**Comment by ssly1997** [comment]
> > > 请记住，每个新的 xcode 版本（如最新的 beta 8）都必须再次进行此更改。我推荐解决方法 2，因为它每次都有效。所以文件 /**Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS113.17.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h** 的第 0 行
> > > 从： 更改为： ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 170000) \` ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 180000) \`
> > 
> > 
> > cool
> 
> @TheMelody WKWebsiteDataStore.h 文件不让修改，怎么办的？

使用默认的文本编辑器

**Comment by VasilisPH** [comment]
> > Thanks @vixez. It worked. But guys I am facing another issue after using the fork. I am new to flutter. So please give your inputs to fix this.
> > > youtube_player_flutter 8.1.2 depends on flutter_inappwebview ^5.7.2+3 and no versions of youtube_player_flutter match >8.1.2 <9.0.0,
> > > youtube_player_flutter ^8.1.2 requires flutter_inappwebview from hosted.
> 
> This is because the package `youtube_player_flutter` you are using doesn't depend on the git dependency you are using.
> 
> You have 2 ways to address this:
> 
>     1. Clone the `youtube_player_flutter` and update its `pubspec.yaml` to use the fork.
> 
>     2. Use [dependency_overrides (not recommended)](https://dart.dev/tools/pub/dependencies#dependency-overrides)

How could we do this ?

**Comment by alipilot149** [comment]
My error:

```
Xcode build done.                                           12.7s
Failed to build iOS app
Error (Xcode): type argument 'nw_proxy_config_t' (aka 'struct nw_proxy_config *') is neither an Objective-C object nor a block type
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h:119:46

Parse Issue (Xcode): Could not build module 'WebKit'
/Users/alipezeshki/StudioProjects/emmanuel/build/ios/Debug-iphonesimulator/flutter_inappwebview/flutter_inappwebview.framework/Headers/flutter_inappwebview-Swift.h:893:8

Could not build the application for the simulator.
Error launching application on iPhone 14.
```



I use this command in my terminal:

open and see /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h

open and edit
sudo nano /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h

```
then window opens, 

#if ((TARGET_OS_OSX && __MAC_OS_X_VERSION_MAX_ALLOWED >= 140000) \
    || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 170000) \
    || (TARGET_OS_WATCH && __WATCH_OS_VERSION_MAX_ALLOWED >= 100000) \
    || (TARGET_OS_TV && __TV_OS_VERSION_MAX_ALLOWED >= 170000))

changed to:

#if ((TARGET_OS_OSX && __MAC_OS_X_VERSION_MAX_ALLOWED >= 140000) \
    || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 180000) \
    || (TARGET_OS_WATCH && __WATCH_OS_VERSION_MAX_ALLOWED >= 100000) \
    || (TARGET_OS_TV && __TV_OS_VERSION_MAX_ALLOWED >= 180000))
```


The problem fixed.

**Comment by durga-nannamsetty** [comment]
If someone still finding difficult to edit "WKWebsiteDataStore.h" file, the workaround is 
1. copy and paste the file somewhere else.
2. edit the file
3. now paste at the original location

**Comment by daironigr** [comment]
> I tried to do Workaround 2 but couldn't edit the file: It said that the file couldn't be written. I have tried with `sudo` as well but no luck. Has anyone experienced the same issue?

Try with `terminal` and sudo. It works for me

**Comment by crasowas** [comment]
There is a generic solution to this, which also applies to other libraries with similar issues.

Append the following code to the end of the `Podfile` file and run the `pod install` command.

```ruby
post_integrate do |installer|
  compiler_flags_key = 'COMPILER_FLAGS'
  project_path = 'Pods/Pods.xcodeproj'
  
  # Backup the Pods.xcodeproj file, this is not necessary.
  # If you don't need it, comment or remove it.
  backup_path = "#{project_path}.bak"
  if !File::exist?(backup_path)
    FileUtils.cp_r(project_path, backup_path)
  end
  
  project = Xcodeproj::Project.open(project_path)
  project.targets.each do |target|
    target.build_phases.each do |build_phase|
      if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
        build_phase.files.each do |file|
          if !file.settings.nil? && file.settings.key?(compiler_flags_key)
            compiler_flags = file.settings[compiler_flags_key]
            file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
          end
        end
      end
    end
  end
  project.save()
end
```

The above solution is achieved by removing the `-DOS OBJECT USE OBJC=0` compiler flag, which is usually used to indicate not to use `ARC`. At present, I don't have enough evidence to prove that the current issue is caused by not using `ARC`, but removing this compiler flag did fix the issue.

So, what does this have to do with `s.platforms`?

When the `Podspec` file is missing `s.platforms` or the version is earlier than iOS 6.0, `CocoaPods` will automatically set the `-DOS_OBJECT_USE_OBJC=0` compiler flag. It follows that setting `s.platform = :ios, '8.0'` also solves the current issue.

```ruby
def compiler_flags_for_consumer(consumer, arc, language)
  flags = consumer.compiler_flags.dup
  if !arc && language == :objc
    flags << '-fno-objc-arc'
  else
    platform_name = consumer.platform_name
    spec_deployment_target = consumer.spec.deployment_target(platform_name)
    if spec_deployment_target.nil? || Version.new(spec_deployment_target) < ENABLE_OBJECT_USE_OBJC_FROM[platform_name]
      flags << '-DOS_OBJECT_USE_OBJC=0'
    end
  end
  if target.inhibit_warnings? && language == :objc
    flags << '-w -Xanalyzer -analyzer-disable-all-checks'
  end
  flags * ' '
end
```

See also: [pod_target_installer.rb](https://github.com/CocoaPods/CocoaPods/blob/master/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb)

**Comment by arpit-1110** [comment]
> I have tried both vscode and vim and no luck. Which one did you use?

I used sudo vim and it worked

**Comment by daironigr** [comment]
> I have tried both vscode and vim and no luck. Which one did you use?

try with sudo, vscode doesn't work for me, I used `nano`

**Comment by abdorizak** [comment]
> > 180000
> 
> > > > I have tried both vscode and vim and no luck. Which one did you use?
> > > 
> > > 
> > > Default TextEditor
> > 
> > 
> > Thanks that helped! I was getting crazy! What a mess updating to iOS17 😅
> 
> How did you save the file after editing it on Textedit because what's i'm get is i don't have permission to save file even after explicitly giving permission to read and write.

i tried with ``` sudo nano file_name.h ``` it works for me

**Comment by Moatamed8** [comment]
After i changed it 180000 how can i solve this ?

Swift Compiler Error (Xcode): Expected value in expression
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h:112:52

Swift Compiler Error (Xcode): Could not build Objective-C module 'WebKit'

Uncategorized (Xcode): Command SwiftCompile failed with a nonzero exit code

Could not build the application for the simulator.
Error launching application on iPhone 15 Pro Max.

**Comment by lanistor** [comment]
@pichillilorenzo Is this problem planned to resolved?

**Comment by abdorizak** [comment]
> After i changed it 180000 how can i solve this ?
> 
> Swift Compiler Error (Xcode): Expected value in expression /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h:112:52
> 
> Swift Compiler Error (Xcode): Could not build Objective-C module 'WebKit'
> 
> Uncategorized (Xcode): Command SwiftCompile failed with a nonzero exit code
> 
> Could not build the application for the simulator. Error launching application on iPhone 15 Pro Max.

it seems like you have issue in the ```WKWebsiteDataStore.h``` at **:112:52** it's something wrong can you share us screenshot of this file ```WKWebsiteDataStore.h```.

**Comment by ivan-lab-pvls** [comment]
Hello, somebody can help me?
![Screenshot 2023-09-25 at 14 22 44](https://github.com/pichillilorenzo/flutter_inappwebview/assets/121606196/dcc3195e-6af6-492f-9a22-97ea050d4a52)
![Screenshot 2023-09-25 at 14 22 57](https://github.com/pichillilorenzo/flutter_inappwebview/assets/121606196/6511cf62-5171-4663-943f-27a4d20cbbf5)

**Comment by Nannoy** [comment]
Noting is working for me here

**Comment by Nannoy** [comment]
I'm planning on changing the ios version to 16.4 maybe that would work

**Comment by Moatamed8** [comment]
> > After i changed it 180000 how can i solve this ?
> > Swift Compiler Error (Xcode): Expected value in expression /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h:112:52
> > Swift Compiler Error (Xcode): Could not build Objective-C module 'WebKit'
> > Uncategorized (Xcode): Command SwiftCompile failed with a nonzero exit code
> > Could not build the application for the simulator. Error launching application on iPhone 15 Pro Max.
> 
> it seems like you have issue in the `WKWebsiteDataStore.h` at **:112:52** it's something wrong can you share us screenshot of this file `WKWebsiteDataStore.h`.

@abdorizak  
<img width="1512" alt="Screenshot 1445-03-10 at 5 28 37 PM" src="https://github.com/pichillilorenzo/flutter_inappwebview/assets/31790122/bfe362aa-e612-48bd-8f2e-1f1a872a8a4c">

**Comment by Nannoy** [comment]
> > > After i changed it 180000 how can i solve this ?
> > > Swift Compiler Error (Xcode): Expected value in expression /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h:112:52
> > > Swift Compiler Error (Xcode): Could not build Objective-C module 'WebKit'
> > > Uncategorized (Xcode): Command SwiftCompile failed with a nonzero exit code
> > > Could not build the application for the simulator. Error launching application on iPhone 15 Pro Max.
> > 
> > 
> > it seems like you have issue in the `WKWebsiteDataStore.h` at **:112:52** it's something wrong can you share us screenshot of this file `WKWebsiteDataStore.h`.
> 
> @abdorizak <img alt="Screenshot 1445-03-10 at 5 28 37 PM" width="1512" src="https://user-images.githubusercontent.com/31790122/270373997-bfe362aa-e612-48bd-8f2e-1f1a872a8a4c.png">

I'm trying a solution I will let you know if it works

**Comment by Hilbert2048** [comment]
for iOS 17 Simulator :  /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk/System/Cryptexes/OS/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h

**Comment by jiadmiftx** [comment]
> Received this from Erick Yim (not showing on the thread?!)
> 
> > WARNING: NOT TESTED
> > workaround 1): in framework xcodeproj Build Phrases -> Compile Sources，select all sources，delete -DOS_OBJECT_USE_OBJC=0 in Complier Flags
> 
> > workaround 2): open /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h, change __IPHONE_OS_VERSION_MAX_ALLOWED to 180000
> 
> workaround 2 has me happy for now! 😊

how can I change the file for workaround 2?

**Comment by JamesBondMine** [comment]
> > 从Erick Yim那里收到这个（没有显示在线程上？！）
> > > 警告：未测试解决方法1）：在框架xcodeproj构建短语->编译源，选择所有源，在编译器标志中删除-DOS_OBJECT_USE_OBJC=0
> > 
> > 
> > > 解决方法2）：打开/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h，将__IPHONE_OS_VERSION_MAX_ALLOWED更改为180000
> > 
> > 
> > 变通方法2让我现在很开心！�
> 
> 无论我做什么，我都无法编辑这个文件。我在终端中尝试了所有东西，默认文本编辑器，它不让我编辑它。



> > Received this from Erick Yim (not showing on the thread?!)
> > > WARNING: NOT TESTED
> > > workaround 1): in framework xcodeproj Build Phrases -> Compile Sources，select all sources，delete -DOS_OBJECT_USE_OBJC=0 in Complier Flags
> > 
> > 
> > > workaround 2): open /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h, change __IPHONE_OS_VERSION_MAX_ALLOWED to 180000
> > 
> > 
> > workaround 2 has me happy for now! 😊
> 
> how can I change the file for workaround 2?

copy this file  to Desktop then edit the file.    after  copy the edited file to the path       over!

**Comment by mduccc** [comment]
I tried `workaround 2` but no luck to me.

![image](https://github.com/pichillilorenzo/flutter_inappwebview/assets/18674411/cd1f02eb-27f2-4fb4-9eb5-e3835b046f63)


> Remember guys that every new xcode build (like latest beta 8) you have to made this change again. I recommend workaround 2, because its working each time. so Line 113 of file **/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h**
> 
> Change from: ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 170000) \` to: ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 180000) \`

**Comment by roshansrashtasoft** [comment]
![Screenshot 2023-09-26 at 12 17 55](https://github.com/pichillilorenzo/flutter_inappwebview/assets/97586645/a77820b1-3e38-4cd9-b38a-f3d2ffaf7f09)

Any Solution I am Stuck

**Comment by crasowas** [comment]
> ![Screenshot 2023-09-26 at 12 17 55](https://user-images.githubusercontent.com/97586645/270561564-a77820b1-3e38-4cd9-b38a-f3d2ffaf7f09.png)
> 
> @currantdev Any Solution I am Stuck 😢 ?

Maybe you can try the following solution.

Append the following code to the end of the `Podfile` file (located in `ios/Podfile`) and run the "pod install" command.

```ruby
post_integrate do |installer|
  compiler_flags_key = 'COMPILER_FLAGS'
  project_path = 'Pods/Pods.xcodeproj'

  project = Xcodeproj::Project.open(project_path)
  project.targets.each do |target|
    target.build_phases.each do |build_phase|
      if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
        build_phase.files.each do |file|
          if !file.settings.nil? && file.settings.key?(compiler_flags_key)
            compiler_flags = file.settings[compiler_flags_key]
            file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
          end
        end
      end
    end
  end
  project.save()
end
```

**Comment by roshansrashtasoft** [comment]
> > ![Screenshot 2023-09-26 at 12 17 55](https://user-images.githubusercontent.com/97586645/270561564-a77820b1-3e38-4cd9-b38a-f3d2ffaf7f09.png)
> > @currantdev Any Solution I am Stuck 😢 ?
> 
> Maybe you can try the following solution.
> 
> Append the following code to the end of the `Podfile` file (located in `ios/Podfile`) and run the "pod install" command.
> 
> ```ruby
> post_integrate do |installer|
>   compiler_flags_key = 'COMPILER_FLAGS'
>   project_path = 'Pods/Pods.xcodeproj'
> 
>   project = Xcodeproj::Project.open(project_path)
>   project.targets.each do |target|
>     target.build_phases.each do |build_phase|
>       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
>         build_phase.files.each do |file|
>           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
>             compiler_flags = file.settings[compiler_flags_key]
>             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
>           end
>         end
>       end
>     end
>   end
>   project.save()
> end
> ```

Any other Solution I have already tried this code.

**Comment by Moatamed8** [comment]
i've solved This problem with perfect solution 
just delete wkwebsitestore.h in Xcode 15 and change it with old one 
i can't upload file but code in this file like this 

`/*
 * Copyright (C) 2014-2017 Apple Inc. All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 * 1. Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 *
 * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
 * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
 * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
 * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
 * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
 * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
 * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
 * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
 * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
 * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 * THE POSSIBILITY OF SUCH DAMAGE.
 */

#import <WebKit/WKFoundation.h>

#import <WebKit/WKWebsiteDataRecord.h>

NS_ASSUME_NONNULL_BEGIN

@class WKHTTPCookieStore;

/*! A WKWebsiteDataStore represents various types of data that a website might
 make use of. This includes cookies, disk and memory caches, and persistent data such as WebSQL,
 IndexedDB databases, and local storage.
 */
WK_EXTERN API_AVAILABLE(macos(10.11), ios(9.0))
@interface WKWebsiteDataStore : NSObject <NSSecureCoding>

/* @abstract Returns the default data store. */
+ (WKWebsiteDataStore *)defaultDataStore;

/** @abstract Returns a new non-persistent data store.
 @discussion If a WKWebView is associated with a non-persistent data store, no data will
 be written to the file system. This is useful for implementing "private browsing" in a web view.
*/
+ (WKWebsiteDataStore *)nonPersistentDataStore;

- (instancetype)new NS_UNAVAILABLE;
- (instancetype)init NS_UNAVAILABLE;

/*! @abstract Whether the data store is persistent or not. */
@property (nonatomic, readonly, getter=isPersistent) BOOL persistent;

/*! @abstract Returns a set of all available website data types. */
+ (NSSet<NSString *> *)allWebsiteDataTypes;

/*! @abstract Fetches data records containing the given website data types.
  @param dataTypes The website data types to fetch records for.
  @param completionHandler A block to invoke when the data records have been fetched.
*/
- (void)fetchDataRecordsOfTypes:(NSSet<NSString *> *)dataTypes completionHandler:(void (^)(NSArray<WKWebsiteDataRecord *> *))completionHandler WK_SWIFT_ASYNC_NAME(dataRecords(ofTypes:));

/*! @abstract Removes website data of the given types for the given data records.
 @param dataTypes The website data types that should be removed.
 @param dataRecords The website data records to delete website data for.
 @param completionHandler A block to invoke when the website data for the records has been removed.
*/
- (void)removeDataOfTypes:(NSSet<NSString *> *)dataTypes forDataRecords:(NSArray<WKWebsiteDataRecord *> *)dataRecords completionHandler:(void (^)(void))completionHandler;

/*! @abstract Removes all website data of the given types that has been modified since the given date.
 @param dataTypes The website data types that should be removed.
 @param date A date. All website data modified after this date will be removed.
 @param completionHandler A block to invoke when the website data has been removed.
*/
- (void)removeDataOfTypes:(NSSet<NSString *> *)dataTypes modifiedSince:(NSDate *)date completionHandler:(void (^)(void))completionHandler;

/*! @abstract Returns the cookie store representing HTTP cookies in this website data store. */
@property (nonatomic, readonly) WKHTTPCookieStore *httpCookieStore API_AVAILABLE(macos(10.13), ios(11.0));

@end

NS_ASSUME_NONNULL_END
`

**Comment by GramGra07** [comment]
Any updates to this being actually fixed in the plugin?

**Comment by IonVillarreal** [comment]
Hi, I have the same issue in Xcode 15 stable release

**Comment by luiszapata6** [comment]
same issue

**Comment by mohyghb** [comment]
same here

**Comment by PankajMobikasa** [comment]
Any solution guys for that issue? I'm stuck to generate a build for latest changes. Please help to fix that issue

**Comment by AresVampire** [comment]
> ```ruby
> post_integrate do |installer|
>   compiler_flags_key = 'COMPILER_FLAGS'
>   project_path = 'Pods/Pods.xcodeproj'
> 
>   project = Xcodeproj::Project.open(project_path)
>   project.targets.each do |target|
>     target.build_phases.each do |build_phase|
>       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
>         build_phase.files.each do |file|
>           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
>             compiler_flags = file.settings[compiler_flags_key]
>             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
>           end
>         end
>       end
>     end
>   end
>   project.save()
> end
> ```

this works for me.

**Comment by afskalsfk** [comment]
> wkwebsitestore

can you tell me where is the file

**Comment by nazar-41** [comment]
> > ![Screenshot 2023-09-26 at 12 17 55](https://user-images.githubusercontent.com/97586645/270561564-a77820b1-3e38-4cd9-b38a-f3d2ffaf7f09.png)
> > @currantdev Any Solution I am Stuck 😢 ?
> 
> Maybe you can try the following solution.
> 
> Append the following code to the end of the `Podfile` file (located in `ios/Podfile`) and run the "pod install" command.
> 
> ```ruby
> post_integrate do |installer|
>   compiler_flags_key = 'COMPILER_FLAGS'
>   project_path = 'Pods/Pods.xcodeproj'
> 
>   project = Xcodeproj::Project.open(project_path)
>   project.targets.each do |target|
>     target.build_phases.each do |build_phase|
>       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
>         build_phase.files.each do |file|
>           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
>             compiler_flags = file.settings[compiler_flags_key]
>             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
>           end
>         end
>       end
>     end
>   end
>   project.save()
> end
> ```

this works for me 👍

**Comment by TeoVogel** [comment]
Same issue, and I'm unable to modify WKWebsiteDataStore.h, not even with sudo, I don't know why 🥲

I have Xcode 15 stable release. Tried targeting an iOS 16.4 simulator but same issue. I can only compile the project for Android.

**Comment by Nannoy** [comment]
Same here, it just shows a white screen and then crashes. Pls someone should help. App works on android emulator

**Comment by aquinoartur** [comment]
same here

**Comment by betheproud** [comment]
> > ![Screenshot 2023-09-26 at 12 17 55](https://user-images.githubusercontent.com/97586645/270561564-a77820b1-3e38-4cd9-b38a-f3d2ffaf7f09.png)
> > @currantdev Any Solution I am Stuck 😢 ?
> 
> Maybe you can try the following solution.
> 
> Append the following code to the end of the `Podfile` file (located in `ios/Podfile`) and run the "pod install" command.
> 
> ```ruby
> post_integrate do |installer|
>   compiler_flags_key = 'COMPILER_FLAGS'
>   project_path = 'Pods/Pods.xcodeproj'
> 
>   project = Xcodeproj::Project.open(project_path)
>   project.targets.each do |target|
>     target.build_phases.each do |build_phase|
>       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
>         build_phase.files.each do |file|
>           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
>             compiler_flags = file.settings[compiler_flags_key]
>             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
>           end
>         end
>       end
>     end
>   end
>   project.save()
> end
> ```

this works for me. thanks. 👍 👍 👍

**Comment by AntonioCardenas** [comment]
https://stackoverflow.com/questions/77189557/type-argument-nw-proxy-config-t-is-neither-an-objective-c-object-nor-a-block-t?noredirect=1#comment136079027_77189557

**Comment by SileoWorkSpace** [comment]
Getting Error


Automatically signing iOS for device deployment using specified development team in Xcode project: 5HDLT27FKL
Upgrading Pods-Runner-frameworks.sh
Xcode build done.                                           20.7s
(lldb) 2023-09-28 13:15:17.942625+0530 Runner[1588:96666] [VERBOSE-2:FlutterDarwinContextMetalImpeller.mm(37)] Using the Impeller rendering backend.
Warning: Unable to create restoration in progress marker file
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
    frame #0: 0x0000000000000000
error: memory read failed for 0x0
Target 0: (Runner) stopped.
Error launching application on iPhone (2).
Exited

**Comment by ahmedhnewa** [comment]
Same issue here.

**Comment by abdorizak** [comment]
Please Upgrade ```cocoapods``` using this ```brew upgrade cocoapods```

**Comment by afskalsfk** [comment]
> > ```ruby
> > post_integrate do |installer|
> >   compiler_flags_key = 'COMPILER_FLAGS'
> >   project_path = 'Pods/Pods.xcodeproj'
> > 
> >   project = Xcodeproj::Project.open(project_path)
> >   project.targets.each do |target|
> >     target.build_phases.each do |build_phase|
> >       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
> >         build_phase.files.each do |file|
> >           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
> >             compiler_flags = file.settings[compiler_flags_key]
> >             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
> >           end
> >         end
> >       end
> >     end
> >   end
> >   project.save()
> > end
> > ```
> 
> 这对我有用。



> > ```ruby
> > post_integrate do |installer|
> >   compiler_flags_key = 'COMPILER_FLAGS'
> >   project_path = 'Pods/Pods.xcodeproj'
> > 
> >   project = Xcodeproj::Project.open(project_path)
> >   project.targets.each do |target|
> >     target.build_phases.each do |build_phase|
> >       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
> >         build_phase.files.each do |file|
> >           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
> >             compiler_flags = file.settings[compiler_flags_key]
> >             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
> >           end
> >         end
> >       end
> >     end
> >   end
> >   project.save()
> > end
> > ```
> 
> this works for me.

Although build was successful, there were problems with the project operation

**Comment by suyashR007** [comment]
> > > 180000
> > 
> > 
> > > > > I have tried both vscode and vim and no luck. Which one did you use?
> > > > 
> > > > 
> > > > Default TextEditor
> > > 
> > > 
> > > Thanks that helped! I was getting crazy! What a mess updating to iOS17 😅
> > 
> > 
> > How did you save the file after editing it on Textedit because what's i'm get is i don't have permission to save file even after explicitly giving permission to read and write.
> 
> you should change the premission for this file but sometimes that is not enough. If that is the case use vim to edit this textfile and use sudo before the vim so you will have: `sudo vim fileName` this way it will work 10000%



> Received this from Erick Yim (not showing on the thread?!)
> 
> > WARNING: NOT TESTED
> > workaround 1): in framework xcodeproj Build Phrases -> Compile Sources，select all sources，delete -DOS_OBJECT_USE_OBJC=0 in Complier Flags
> 
> > workaround 2): open /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h, change __IPHONE_OS_VERSION_MAX_ALLOWED to 180000
> 
> workaround 2 has me happy for now! 😊

This worked for me tho later when i tried to run I found it was throwing error
Swift Compiler Error (Xcode): Using bridging headers with module interfaces is unsupported


Uncategorized (Xcode): Command SwiftDriver emitted errors but did not return a nonzero exit code to indicate failure


Error (Xcode): Build input file cannot be found: '/Users/***/ios/Runner/Info.plist'. Did you forget to declare
this file as an output of a script phase or custom build rule which produces it?


Could not build the application for the simulator.
Error launching application on iPhone 15 Pro Max.

**Comment by AntonioCardenas** [comment]
> Remember guys that every new xcode build (like latest beta 8) you have to made this change again. I recommend workaround 2, because its working each time. so Line 113 of file **/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h**
> 
> Change from: ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 170000) \` to: ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 180000) \`

This work around works for my project also I made a stack overflow post fell free to add any workarounds .

https://stackoverflow.com/questions/77189557/type-argument-nw-proxy-config-t-is-neither-an-objective-c-object-nor-a-block-t?noredirect=1#comment136079027_77189557

**Comment by christoph-fer** [comment]
> > ![Screenshot 2023-09-26 at 12 17 55](https://user-images.githubusercontent.com/97586645/270561564-a77820b1-3e38-4cd9-b38a-f3d2ffaf7f09.png)
> > @currantdev Any Solution I am Stuck 😢 ?
> 
> Maybe you can try the following solution.
> 
> Append the following code to the end of the `Podfile` file (located in `ios/Podfile`) and run the "pod install" command.
> 
> ```ruby
> post_integrate do |installer|
>   compiler_flags_key = 'COMPILER_FLAGS'
>   project_path = 'Pods/Pods.xcodeproj'
> 
>   project = Xcodeproj::Project.open(project_path)
>   project.targets.each do |target|
>     target.build_phases.each do |build_phase|
>       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
>         build_phase.files.each do |file|
>           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
>             compiler_flags = file.settings[compiler_flags_key]
>             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
>           end
>         end
>       end
>     end
>   end
>   project.save()
> end
> ```

Thanks a ton! This worked for me! 😭

**Comment by christoph-fer** [comment]
> > ![Screenshot 2023-09-26 at 12 17 55](https://user-images.githubusercontent.com/97586645/270561564-a77820b1-3e38-4cd9-b38a-f3d2ffaf7f09.png)
> > @currantdev Any Solution I am Stuck 😢 ?
> 
> Maybe you can try the following solution.
> 
> Append the following code to the end of the `Podfile` file (located in `ios/Podfile`) and run the "pod install" command.
> 
> ```ruby
> post_integrate do |installer|
>   compiler_flags_key = 'COMPILER_FLAGS'
>   project_path = 'Pods/Pods.xcodeproj'
> 
>   project = Xcodeproj::Project.open(project_path)
>   project.targets.each do |target|
>     target.build_phases.each do |build_phase|
>       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
>         build_phase.files.each do |file|
>           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
>             compiler_flags = file.settings[compiler_flags_key]
>             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
>           end
>         end
>       end
>     end
>   end
>   project.save()
> end
> ```

This worked for me as well! Thanks a ton! :)

For the ones this doesn't work try deleting your `podfile.lock` and run `cd ios pod install`

**Comment by SileoWorkSpace** [comment]
After upgrade cocoa pods getting below mentioned error

Automatically signing iOS for device deployment using specified development team in Xcode project: 5HDLT2798K
Xcode build done.                                           18.6s
(lldb) 2023-09-29 13:34:03.993105+0530 Runner[4066:283159] Warning: Unable to create restoration in progress marker file
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
    frame #0: 0x0000000000000000
error: memory read failed for 0x0
Target 0: (Runner) stopped.
Error launching application on iPhone (2).

**Comment by fauzannoorazhar** [comment]
> https://stackoverflow.com/questions/77189557/type-argument-nw-proxy-config-t-is-neither-an-objective-c-object-nor-a-block-t?noredirect=1#comment136079027_77189557

Im still have an issue, following for this. Any update?

**Comment by fauzannoorazhar** [comment]
> > > ![Screenshot 2023-09-26 at 12 17 55](https://user-images.githubusercontent.com/97586645/270561564-a77820b1-3e38-4cd9-b38a-f3d2ffaf7f09.png)
> > > @currantdev Any Solution I am Stuck 😢 ?
> > 
> > 
> > Maybe you can try the following solution.
> > Append the following code to the end of the `Podfile` file (located in `ios/Podfile`) and run the "pod install" command.
> > ```ruby
> > post_integrate do |installer|
> >   compiler_flags_key = 'COMPILER_FLAGS'
> >   project_path = 'Pods/Pods.xcodeproj'
> > 
> >   project = Xcodeproj::Project.open(project_path)
> >   project.targets.each do |target|
> >     target.build_phases.each do |build_phase|
> >       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
> >         build_phase.files.each do |file|
> >           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
> >             compiler_flags = file.settings[compiler_flags_key]
> >             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
> >           end
> >         end
> >       end
> >     end
> >   end
> >   project.save()
> > end
> > ```
> 
> This worked for me as well! Thanks a ton! :)
> 
> For the ones this doesn't work try deleting your `podfile.lock` and run `cd ios pod install`

Im still cannot archive and build on Emulator iPhone 15, any update or detail to fix?

**Comment by CodeSadhu** [comment]
> > ![Screenshot 2023-09-26 at 12 17 55](https://user-images.githubusercontent.com/97586645/270561564-a77820b1-3e38-4cd9-b38a-f3d2ffaf7f09.png)
> > @currantdev Any Solution I am Stuck 😢 ?
> 
> Maybe you can try the following solution.
> 
> Append the following code to the end of the `Podfile` file (located in `ios/Podfile`) and run the "pod install" command.
> 
> ```ruby
> post_integrate do |installer|
>   compiler_flags_key = 'COMPILER_FLAGS'
>   project_path = 'Pods/Pods.xcodeproj'
> 
>   project = Xcodeproj::Project.open(project_path)
>   project.targets.each do |target|
>     target.build_phases.each do |build_phase|
>       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
>         build_phase.files.each do |file|
>           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
>             compiler_flags = file.settings[compiler_flags_key]
>             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
>           end
>         end
>       end
>     end
>   end
>   project.save()
> end
> ```

Dude. I've been scratching my brain over this error ever since I upgraded to Sonoma stable. I had almost started regretting the decision to upgrade till I found this. Thanks a million!
I would like to know why this is happening, btw. I'm not an iOS developer.

**Comment by Manguelo** [comment]
Is there any update as to when a fix for this will be live?

**Comment by fauzannoorazhar** [comment]
> > > ![Screenshot 2023-09-26 at 12 17 55](https://user-images.githubusercontent.com/97586645/270561564-a77820b1-3e38-4cd9-b38a-f3d2ffaf7f09.png)
> > > @currantdev Any Solution I am Stuck 😢 ?
> > 
> > 
> > Maybe you can try the following solution.
> > Append the following code to the end of the `Podfile` file (located in `ios/Podfile`) and run the "pod install" command.
> > ```ruby
> > post_integrate do |installer|
> >   compiler_flags_key = 'COMPILER_FLAGS'
> >   project_path = 'Pods/Pods.xcodeproj'
> > 
> >   project = Xcodeproj::Project.open(project_path)
> >   project.targets.each do |target|
> >     target.build_phases.each do |build_phase|
> >       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
> >         build_phase.files.each do |file|
> >           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
> >             compiler_flags = file.settings[compiler_flags_key]
> >             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
> >           end
> >         end
> >       end
> >     end
> >   end
> >   project.save()
> > end
> > ```
> 
> This worked for me as well! Thanks a ton! :)
> 
> For the ones this doesn't work try deleting your `podfile.lock` and run `cd ios pod install`

Thanks, now is solved. 

```
Flutter 3.7.9 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 62bd79521d (6 months ago) • 2023-03-30 10:59:36 -0700
Engine • revision ec975089ac
Tools • Dart 2.19.6 • DevTools 2.20.1

Pod Version : 1.13.0

Xcode 15.0
Build version 15A240d
```

**Comment by applevinc** [comment]
> > ![Screenshot 2023-09-26 at 12 17 55](https://user-images.githubusercontent.com/97586645/270561564-a77820b1-3e38-4cd9-b38a-f3d2ffaf7f09.png)
> > @currantdev Any Solution I am Stuck 😢 ?
> 
> Maybe you can try the following solution.
> 
> Append the following code to the end of the `Podfile` file (located in `ios/Podfile`) and run the "pod install" command.
> 
> ```ruby
> post_integrate do |installer|
>   compiler_flags_key = 'COMPILER_FLAGS'
>   project_path = 'Pods/Pods.xcodeproj'
> 
>   project = Xcodeproj::Project.open(project_path)
>   project.targets.each do |target|
>     target.build_phases.each do |build_phase|
>       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
>         build_phase.files.each do |file|
>           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
>             compiler_flags = file.settings[compiler_flags_key]
>             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
>           end
>         end
>       end
>     end
>   end
>   project.save()
> end
> ```

Please share a fix if the project has build configurations

**Comment by kahdichienja** [comment]
> > I tried to do Workaround 2 but couldn't edit the file: It said that the file couldn't be written. I have tried with `sudo` as well but no luck. Has anyone experienced the same issue?
> 
> Using a text editor instead of Xcode

use terimal with nano, ensure-> sudo nano /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h


then you machine will prompt you to add previlages, do that then relaunch the terminal, redu the command -> sudo nano /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h

This time it should work

**Comment by egor-slatvinskiy-idf** [comment]
> > > ![Screenshot 2023-09-26 at 12 17 55](https://user-images.githubusercontent.com/97586645/270561564-a77820b1-3e38-4cd9-b38a-f3d2ffaf7f09.png)
> > > @currantdev Any Solution I am Stuck 😢 ?
> > 
> > 
> > Maybe you can try the following solution.
> > Append the following code to the end of the `Podfile` file (located in `ios/Podfile`) and run the "pod install" command.
> > ```ruby
> > post_integrate do |installer|
> >   compiler_flags_key = 'COMPILER_FLAGS'
> >   project_path = 'Pods/Pods.xcodeproj'
> > 
> >   project = Xcodeproj::Project.open(project_path)
> >   project.targets.each do |target|
> >     target.build_phases.each do |build_phase|
> >       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
> >         build_phase.files.each do |file|
> >           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
> >             compiler_flags = file.settings[compiler_flags_key]
> >             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
> >           end
> >         end
> >       end
> >     end
> >   end
> >   project.save()
> > end
> > ```
> 
> Please share a fix if the project has build configurations

the same

**Comment by GramGra07** [comment]
> > > ![Screenshot 2023-09-26 at 12 17 55](https://user-images.githubusercontent.com/97586645/270561564-a77820b1-3e38-4cd9-b38a-f3d2ffaf7f09.png)
> > > @currantdev Any Solution I am Stuck 😢 ?
> > 
> > 
> > Maybe you can try the following solution.
> > Append the following code to the end of the `Podfile` file (located in `ios/Podfile`) and run the "pod install" command.
> > ```ruby
> > post_integrate do |installer|
> >   compiler_flags_key = 'COMPILER_FLAGS'
> >   project_path = 'Pods/Pods.xcodeproj'
> > 
> >   project = Xcodeproj::Project.open(project_path)
> >   project.targets.each do |target|
> >     target.build_phases.each do |build_phase|
> >       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
> >         build_phase.files.each do |file|
> >           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
> >             compiler_flags = file.settings[compiler_flags_key]
> >             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
> >           end
> >         end
> >       end
> >     end
> >   end
> >   project.save()
> > end
> > ```
> 
> Please share a fix if the project has build configurations

this worked for me

**Comment by Navdeep-hunch** [comment]
Use the latest version of flutter_inappwebview. ie.  flutter_inappwebview: ^5.8.0. 
They have fixed the issue in the latest version

**Comment by pichillilorenzo** [comment]
Released new version `5.8.0` and `6.0.0-beta.25` with the fix!

**Comment by reysagar** [comment]
> > ```ruby
> > post_integrate do |installer|
> >   compiler_flags_key = 'COMPILER_FLAGS'
> >   project_path = 'Pods/Pods.xcodeproj'
> > 
> >   project = Xcodeproj::Project.open(project_path)
> >   project.targets.each do |target|
> >     target.build_phases.each do |build_phase|
> >       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
> >         build_phase.files.each do |file|
> >           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
> >             compiler_flags = file.settings[compiler_flags_key]
> >             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
> >           end
> >         end
> >       end
> >     end
> >   end
> >   project.save()
> > end
> > ```

> this works for me.

Getting this error after adding the above code on the podfile, any idea how to resolve it?

```
Lexical or Preprocessor Issue (Xcode): 'Flutter/Flutter.h' file not found
/Users/evo/.pub-cache/hosted/pub.dev/sqflite-2.2.4+1/ios/Classes/SqfliteImport.h:10:8
```

**Comment by reysagar** [comment]
```
post_integrate do |installer|
  compiler_flags_key = 'COMPILER_FLAGS'
  project_path = 'Pods/Pods.xcodeproj'

  project = Xcodeproj::Project.open(project_path)
  project.targets.each do |target|
    target.build_phases.each do |build_phase|
      if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
        build_phase.files.each do |file|
          if !file.settings.nil? && file.settings.key?(compiler_flags_key)
            compiler_flags = file.settings[compiler_flags_key]
            file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
          end
        end
      end
    end
  end
  project.save()
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
    target.build_configurations.each do |config|
      config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '13.0'
    end
  end
end
```

This worked for me, just add a combination of the above code on the podfile.

```
Flutter 3.13.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision e1e47221e8 (6 weeks ago) • 2023-08-22 21:43:18 -0700
Engine • revision b20183e040
Tools • Dart 3.1.0 • DevTools 2.25.0
Cocopods 1.13.0
```

**Comment by wdtheprovider** [comment]
> Use the latest version of flutter_inappwebview. ie. flutter_inappwebview: ^5.8.0. They have fixed the issue in the latest version

This helped me

**Comment by tripura4u** [comment]
Please dont close this topic.

Team,

I am facing the same issue. Tried all the solutions, does not work 

changed in Podfile - does not work
Editing the file - I am not able to save (tried with vim, nano, vscode xcode)
Is there anything else that can be done

**Comment by erickyim** [comment]
> Please dont close this topic.
> 
> Team,
> 
> I am facing the same issue. Tried all the solutions, does not work
> 
> changed in Podfile - does not work Editing the file - I am not able to save (tried with vim, nano, vscode xcode) Is there anything else that can be done

You need root permission, try sudo vim, or vscode (will prompt root permission when u save system files).

**Comment by zierka** [comment]
> Released new version `5.8.0` and `6.0.0-beta.25` with the fix!

This should fix this issue. Make sure to do a `pod install` in the `ios` folder after doing the `pub get`. If you get an error, do `pod repo update` first.

**Comment by hammasali** [comment]
> > > > ![Screenshot 2023-09-26 at 12 17 55](https://user-images.githubusercontent.com/97586645/270561564-a77820b1-3e38-4cd9-b38a-f3d2ffaf7f09.png)
> > > > @currantdev Any Solution I am Stuck 😢 ?
> > > 
> > > 
> > > Maybe you can try the following solution.
> > > Append the following code to the end of the `Podfile` file (located in `ios/Podfile`) and run the "pod install" command.
> > > ```ruby
> > > post_integrate do |installer|
> > >   compiler_flags_key = 'COMPILER_FLAGS'
> > >   project_path = 'Pods/Pods.xcodeproj'
> > > 
> > >   project = Xcodeproj::Project.open(project_path)
> > >   project.targets.each do |target|
> > >     target.build_phases.each do |build_phase|
> > >       if build_phase.is_a?(Xcodeproj::Project::Object::PBXSourcesBuildPhase)
> > >         build_phase.files.each do |file|
> > >           if !file.settings.nil? && file.settings.key?(compiler_flags_key)
> > >             compiler_flags = file.settings[compiler_flags_key]
> > >             file.settings[compiler_flags_key] = compiler_flags.gsub(/-DOS_OBJECT_USE_OBJC=0\s*/, '')
> > >           end
> > >         end
> > >       end
> > >     end
> > >   end
> > >   project.save()
> > > end
> > > ```
> > 
> > 
> > Please share a fix if the project has build configurations
> 
> I am facing this error <img alt="Screenshot 2023-10-10 at 1 09 11 PM" width="1016" src="https://user-images.githubusercontent.com/52609580/273824529-c8a7222e-dec6-4570-8d6d-74dba49e4cd6.png">

I fixed this by checking ENABLE_USER_SCRIPT_SANDBOXING to No in the project's build settings. It works perfectly,

**Comment by olintosynergy** [comment]
> Remember guys that every new xcode build (like latest beta 8) you have to made this change again. I recommend workaround 2, because its working each time. so Line 113 of file **/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h**
> 
> Change from: ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 170000) \` to: ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 180000) \`



> Remember guys that every new xcode build (like latest beta 8) you have to made this change again. I recommend workaround 2, because its working each time. so Line 113 of file **/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h**
> 
> Change from: ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 170000) \` to: ` || ((TARGET_OS_IOS || TARGET_OS_MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 180000) \`

For me is Good!! Tks!

**Comment by erickyim** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
